### ISSUE TITLE
pseudocolor debug=None

### ISSUE THREAD

**Issue by dschneiderch** [danforthcenter/plantcv#482]
**Describe the bug**
I know there was a bunch of iteration on the plotting output with pseudocolor but i'm running into problems running on a headless linux server running centos. 

First, I was getting an error
`qt.qpa.screen: QXcbConnection: Could not connect to display`
which I tried to resolve with `export QT_QPA_PLATFORM='offscreen'`  per https://github.com/ipython/ipython/issues/10627 even though it doesn't sounds like it should be necessary.
Then I tried adding
`import matplotlib; matplotlib.use('agg')`
and it throws lots of warnings because it cant show figures with this backend. It's coming from pseudocolor
```
/home/dominik/Applications/miniconda3/envs/test-environment/lib/python3.7/site-packages/plantcv-3.6.2+8.g640a9
68-py3.7.egg/plantcv/plantcv/visualize/pseudocolor.py:142: UserWarning: Matplotlib is currently using agg, whi
ch is a non-GUI backend, so cannot show the figure.
  plt.show(block=False)
```

I guess it's 
```
if debug == None:
    plt.show(block=False)
```

@HaleySchuhl  @nfahlgren  Is there a reason you are showing the plot if debug = None?

**Comment by nfahlgren** [comment]
Ah okay, that explains why the display is set

**Comment by dschneiderch** [comment]
I just tried it while running an x server on my computer but I get the same error message

**Comment by dschneiderch** [comment]
Looks like it works without x forwarding.

**Comment by nfahlgren** [comment]
If I log into our system with X11 forwarding I get an error too, but it's not the same as yours (mine is a libGL error). So this probably won't work but I fixed it on my end by setting this in my environment: `export LIBGL_ALWAYS_INDIRECT=1`

We can look at the plotting commands again potentially. We are not trying to generate a plot that gets displayed but the issue was that we didn't see a way to make the plot without triggering it to display. Not an issue when the backend is Agg or in Jupyter, but obviously can cause issues otherwise.

The other fix that works is doing this before importing plantcv:

```python
import matplotlib
matplotlib.use("Agg")
```

Matplotlib only lets you set the backend once, so if you do it before PlantCV tries you will override it. It will print out a warning but it's not an error.

**Comment by dschneiderch** [comment]
@nfahlgren  I just tested a graphics-device-error branch where I removed plt.show(block=False) if Debug==None and it ran your test script above fine, including the output. was something else not working?

**Comment by dschneiderch** [comment]
i should clarify, it still didn't work with X11 Forwarding but it worked without the forwarding

**Comment by killerpanda101** [comment]
so i am using plant-cv on the raspberry-pi to try to build an plant growth tracking system and did not want to display the pseudo-color image to be displayed and only to be written to a file.

import matplotlib
matplotlib.use("Agg")

before importing plant-cv seems to do the trick

**Comment by nfahlgren** [comment]
@killerpanda101 this brings up a good point. We set the plotting backend automatically when there is/is not a display device, but when you have a display and do not want to use it our assumption is wrong. We set it on import so I am not sure how easy it would be to make it set-able by the user, but at the very least we should update the documentation to suggest importing matplotlib and setting the backend prior to importing PlantCV when you want to disable the display.
